core/storage/feedback/gae_models.py

```diff
@@ -420,6 +421,400 @@ def get_multi(cls, user_id, thread_ids):
+
+
+class GeneralFeedbackMessageModel(base_models.BaseModel):
+    """Feedback messages. One or more of these messages make a thread.
+
+    The id of instances of this class has the form [THREAD_ID].[MESSAGE_ID]
+    """
+    # ID corresponding to an entry of FeedbackThreadModel.
+    thread_id = ndb.StringProperty(required=True, indexed=True)
+    # 0-based sequential numerical ID. Sorting by this field will create the
+    # thread in chronological order.
+    message_id = ndb.IntegerProperty(required=True, indexed=True)
+    # ID of the user who posted this message. This may be None if the feedback
+    # was given anonymously by a learner.
+    author_id = ndb.StringProperty(indexed=True)
+    # New thread status. Must exist in the first message of a thread. For the
+    # rest of the thread, should exist only when the status changes.
+    updated_status = ndb.StringProperty(choices=STATUS_CHOICES, indexed=True)
+    # New thread subject. Must exist in the first message of a thread. For the
+    # rest of the thread, should exist only when the subject changes.
+    updated_subject = ndb.StringProperty(indexed=False)
+    # Message text. Allowed not to exist (e.g. post only to update the status).
+    text = ndb.StringProperty(indexed=False)
+    # Whether the incoming message is received by email (as opposed to via
+    # the web).
+    received_via_email = ndb.BooleanProperty(default=False, indexed=True)
+    # TODO (nithesh): Overriding last_updated of base model for migrating
+    # instances from old model to the new one. To be removed after migration.
+    last_updated = ndb.DateTimeProperty()
+
+    @classmethod
+    def _generate_id(cls, thread_id, message_id):
+        """Generates full message ID given the thread ID and message ID.
+
+        Args:
+            thread_id: str. Thread ID of the thread to which the message
+                belongs.
+            message_id: int. Message ID of the message.
+
+        Returns:
+            str. Full message ID.
+        """
+        return '.'.join([thread_id, str(message_id)])
+
+    @property
+    def entity_id(self):
+        """Returns the entity_id corresponding to this thread instance.
+
+        Returns:
+            str. The entity_id.
+        """
+        return self.id.split('.')[1]
+
+    @property
+    def entity_type(self):
+        """Returns the entity_type corresponding to this thread instance.
+
+        Returns:
+            str. The entity_type.
+        """
+        return self.id.split('.')[0]
+
+
+    def get_thread_subject(self):
+        """Returns the subject of the thread corresponding to this
+        GeneralFeedbackMessageModel instance.
+
+        Returns:
+            str. The subject of the thread.
+        """
+        return GeneralFeedbackThreadModel.get_by_id(self.thread_id).subject
+
+    @classmethod
+    def create(cls, thread_id, message_id):
+        """Creates a new FeedbackMessageModel entry.
```

Comment:
You need to update the names of the classes in these docstrings, here and below.

Reply:
Done